simple-facebook-instant-articles.php

```diff
@@ -393,6 +395,42 @@ public function render_pull_quotes( \DOMDocument &$dom, \DOMXPath &$xpath ) {
 	}
 
 	/**
+	 * Renders images into FB AI format that don't have <figure> element as parent.
+	 * Consider <img> with parent <figure> already been converted to FB IA format.
+	 *
+	 * Ref: https://developers.facebook.com/docs/instant-articles/reference/image
+	 *
+	 * @param DOMDocument $dom   DOM object generated for post content.
+	 * @param DOMXPath    $xpath XPATH object generated for post content.
+	 */
+	public function render_images( \DOMDocument &$dom, \DOMXPath &$xpath ) {
+
+		// Images - with parent that's not <figure>.
+		foreach ( $xpath->query( '//img[not(parent::figure)]' ) as $node ) {
+
+			// Wrap <img> into <figure>.
+			$image_html = $this->get_html_for_node( $node, 'xml', true );
```

Comment:
Ah I see.... we need valid XML in order to use appendXML. Actually I don't think we need to do this at all. Lets just create a `figure` element, replace the image with it, and append the image to the `figure`.